scipy/linalg/tests/test_solvers.py

```diff
@@ -135,40 +135,182 @@ def test_cases(self):
 class TestSolveDiscreteARE(TestCase):
 
     cases = [
-        # Difficult from a numerical standpoint, again from Laub, A. J.
-        # (http://dspace.mit.edu/bitstream/handle/1721.1/1301/R-0859-05666488.pdf)
-        (np.matrix([[4, 3], [-9.0/2.0, -7.0/2.0]]),
-         np.matrix([[1,], [-1,]]),
-         np.matrix([[9, 6], [6, 4]]),
-         np.matrix([[1,],])),
-        # Another example from Laub
-        (np.matrix([[0.9512, 0], [0, 0.9048]]),
-         np.matrix([[4.877, 4.877], [-1.1895, 3.569]]),
-         np.matrix([[0.005, 0],[0, 0.02]]),
-         np.matrix([[1.0/3.0, 0],[0, 3]])),
+        # Darex examples taken from (with default parameters):
+        # [1] P.BENNER, A.J. LAUB, V. MEHRMANN: 'A Collection of Benchmark 
+        #     Examples for the Numerical Solution of Algebraic Riccati 
+        #     Equations II: Discrete-Time Case', Tech. Report SPC 95_23, 
+        #     Fak. f. Mathematik, TU Chemnitz-Zwickau (Germany), 1995.
+        # [2] T. GUDMUNDSSON, C. KENNEY, A.J. LAUB: 'Scaling of the 
+        #     Discrete-Time Algebraic Riccati Equation to Enhance Stability 
+        #     of the Schur Solution Method', IEEE Trans.Aut.Cont., vol.37(4) 
+        # 
         # Complex a; real b, q, r
-        (np.matrix([[2, 1-2j], [0, -3j]]),
-         np.matrix([[0,], [1,]]),
-         np.matrix([[1, 0], [0, 2]]),
-         np.matrix([[1,],])),
+        (np.array([[2, 1-2j], [0, -3j]]),
+         np.array([[0,], [1,]]),
+         np.array([[1, 0], [0, 2]]),
+         np.array([[1,],])),
         # Real a, q, r; complex b
-        (np.matrix([[2, 1], [0, -1]]),
-         np.matrix([[-2j,], [1j,]]),
-         np.matrix([[1, 0], [0, 2]]),
-         np.matrix([[1,],])),
-        # Real a, b; complex q, r
-        (np.matrix([[3, 1], [0, -1]]),
-         np.matrix([[1, 2], [1, 3]]),
-         np.matrix([[1, -3j], [1-1j, 2]]),
-         np.matrix([[-2j, 2], [1j, 3]])),
+        (np.array([[2, 1], [0, -1]]),
+         np.array([[-2j,], [1j,]]),
+         np.array([[1, 0], [0, 2]]),
+         np.array([[1,],])),
+        # Real a, b; complex q, r 
+        (np.array([[3, 1], [0, -1]]),
+         np.array([[1, 2], [1, 3]]),
+         np.array([[1, 1+1j], [1-1j, 2]]),
+         np.array([[2, -2j], [2j, 3]])),
+        # User-reported 
```

Comment:
For got to mention this: reference the original github issue here so it's easy to find later.